pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -525,7 +548,10 @@ LoopOnIterator:
 		}
 
 		pool.Go(func(ctx context.Context) error {
-			atomic.AddUint32(resolutionMetadata.DispatchCount, 1)
+			_ = atomic.AddUint32(resolutionMetadata.DispatchCount, 1)
+			if c.dispatchThrottlerConfig.Enabled {
+				c.throttle(ctx, resolutionMetadata)
+			}
 			return c.execute(ctx, &ReverseExpandRequest{
```

Comment:
instead of having this line in 2 places why can't we put

```
if c.dispatchThrottlerConfig.Enabled {
				c.throttle(ctx, resolutionMetadata)
			}
```

inside `c.execute`, directly?